api/resolvers/wallet.js

```diff
@@ -380,18 +382,10 @@ export default {
         throw new GraphQLError('bad hmac', { extensions: { code: 'FORBIDDEN' } })
       }
       await cancelHodlInvoice({ id: hash, lnd })
-      const inv = await serialize(
-        models.invoice.update({
-          where: {
-            hash
-          },
-          data: {
-            cancelled: true
-          }
-        }),
-        { models }
-      )
-      return inv
+      await models.$executeRaw`
+        INSERT INTO pgboss.job (name, data, retrylimit, retrybackoff, priority)
+          VALUES ('checkInvoice', jsonb_build_object('hash', ${hash}), 21, true, 100)`
+      return await models.invoice.findFirst({ where: { hash } })
```

Comment:
So that we can reuse the existing invoice checking code.

Reply:
Why not import `checkInvoice` and run it ourselves?

Like this, this means that the invoice isn't actually cancelled yet when the mutation returns.

Even the response would indicate that. So a hypothetical user interface where we have a cancel button next to the invoice state, the invoice state wouldn't update on cancel (since it's not cancelled yet).

We don't have this anywhere yet afaict but I considered being able to cancel deposits manually (instead of letting them expire) in the past.

Reply:
Good rec!